In [1]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns 
import numpy as np

___

# Music Recommendation Training Dataset Overview

___

In [19]:
#Store music recommendation training dataset file path in variable for readability
train_path = ("/Users/sa14/Desktop/Music Recommendation Algorithm with Unsupervised Learning/data/music_rec_trainset.csv")

#Load music recommendation training dataset csv as a dataframe
df = pd.read_csv(train_path)

In [3]:
#Get the first 10 rows of the music recommendation training dataframe
df.head(10)

Unnamed: 0           artist_name  \
0           0                mukesh   
1           4         frankie laine   
2           6           johnnie ray   
3          10           pérez prado   
4          12  giorgos papadopoulos   
5          14            perry como   
6          15             freestyle   
7          17         johnny mathis   
8          20   stélios kazantzídis   
9          23   stélios kazantzídis   

                                          track_name  release_date genre  \
0                               mohabbat bhi jhoothi          1950   pop   
1                                          i believe          1950   pop   
2                                                cry          1950   pop   
3                                           patricia          1950   pop   
4                                 apopse eida oneiro          1950   pop   
5  round and round (with mitchell ayres and his o...          1950   pop   
6                         opm medley: when i met you          1950   pop   
7                             it's not for me to say          1950   pop   
8                           klapse me mana klapse me          1950   pop   
9                                  finito la mouzika          1950   pop   

                                              lyrics  len    dating  violence  \
0  hold time feel break feel untrue convince spea...   95  0.000598  0.063746   
1  believe drop rain fall grow believe darkest ni...   51  0.035537  0.096777   
2  sweetheart send letter goodbye secret feel bet...   24  0.002770  0.002770   
3  kiss lips want stroll charm mambo chacha merin...   54  0.048249  0.001548   
4  till darling till matter know till dream live ...   48  0.001350  0.001350   
5  convoy light dead ahead merchantmen trump dies...   98  0.001053  0.420685   
6  piece mindin world knowin life come bring give...  179  0.000483  0.000483   
7  care moment hold fast press lips dream heaven ...   21  0.002506  0.002506   
8  lonely night surround power read mind hour nig...   30  0.083537  0.002024   
9  tear heart seat stay awhile tear heart game st...   61  0.001120  0.102548   

   world/life  ...  communication   obscene     music  movement/places  \
0    0.000598  ...       0.263751  0.000598  0.039288         0.000598   
1    0.443435  ...       0.001284  0.001284  0.118034         0.001284   
2    0.002770  ...       0.250668  0.002770  0.323794         0.002770   
3    0.001548  ...       0.001548  0.001548  0.001548         0.129250   
4    0.417772  ...       0.001350  0.001350  0.001350         0.001350   
5    0.001053  ...       0.001053  0.001053  0.001053         0.001053   
6    0.371817  ...       0.384703  0.000483  0.000483         0.000483   
7    0.336056  ...       0.002506  0.002506  0.002506         0.002506   
8    0.002024  ...       0.167783  0.002024  0.002024         0.002024   
9    0.001120  ...       0.091572  0.001120  0.001120         0.105752   

   light/visual perceptions  family/spiritual   sadness  feelings       topic  \
0                  0.000598          0.000598  0.380299  0.117175     sadness   
1                  0.212681          0.051124  0.001284  0.001284  world/life   
2                  0.002770          0.002770  0.002770  0.225422       music   
3                  0.001548          0.001548  0.225889  0.001548    romantic   
4                  0.001350          0.029755  0.068800  0.001350    romantic   
5                  0.172173          0.001053  0.128292  0.001053    violence   
6                  0.044355          0.000483  0.000483  0.123150  world/life   
7                  0.386887          0.002506  0.062602  0.002506  world/life   
8                  0.002024          0.002024  0.002024  0.031600    romantic   
9                  0.001120          0.001120  0.630507  0.001120     sadness   

   age  
0  1.0  
1  1.0  
2  1.0  
3  1.0  
4  1.0  
5  1.0  
6  1.0  
7  1.0  
8  1.0  
9  1.0  

[10 rows x 24 columns]

In [14]:
#Remove "Unnamed:0" column from dataframe
df.drop(df.filter(regex="Unnamed").columns, axis=1, inplace=True)

#Verify "Unnamed:0" column removal
df.head()


artist_name            track_name  release_date genre  \
0                mukesh  mohabbat bhi jhoothi          1950   pop   
1         frankie laine             i believe          1950   pop   
2           johnnie ray                   cry          1950   pop   
3           pérez prado              patricia          1950   pop   
4  giorgos papadopoulos    apopse eida oneiro          1950   pop   

                                              lyrics  len    dating  violence  \
0  hold time feel break feel untrue convince spea...   95  0.000598  0.063746   
1  believe drop rain fall grow believe darkest ni...   51  0.035537  0.096777   
2  sweetheart send letter goodbye secret feel bet...   24  0.002770  0.002770   
3  kiss lips want stroll charm mambo chacha merin...   54  0.048249  0.001548   
4  till darling till matter know till dream live ...   48  0.001350  0.001350   

   world/life  night/time  ...  communication   obscene     music  \
0    0.000598    0.000598  ...       0.263751  0.000598  0.039288   
1    0.443435    0.001284  ...       0.001284  0.001284  0.118034   
2    0.002770    0.002770  ...       0.250668  0.002770  0.323794   
3    0.001548    0.001548  ...       0.001548  0.001548  0.001548   
4    0.417772    0.001350  ...       0.001350  0.001350  0.001350   

   movement/places  light/visual perceptions  family/spiritual   sadness  \
0         0.000598                  0.000598          0.000598  0.380299   
1         0.001284                  0.212681          0.051124  0.001284   
2         0.002770                  0.002770          0.002770  0.002770   
3         0.129250                  0.001548          0.001548  0.225889   
4         0.001350                  0.001350          0.029755  0.068800   

   feelings       topic  age  
0  0.117175     sadness  1.0  
1  0.001284  world/life  1.0  
2  0.225422       music  1.0  
3  0.001548    romantic  1.0  
4  0.001350    romantic  1.0  

[5 rows x 23 columns]

In [15]:
#Get the count of rows and columns for the dataset
df.shape

(28362, 23)

**This dataset has 28,362 rows and 23 columns after removing the "Unnamed: 0" column**

In [18]:
#Get overall count of null/NaN values in the dataset
null_count = df.isnull().sum()
print("Null Values by Column \n",null_count)

Null Values by Column 
 artist_name                 0
track_name                  0
release_date                0
genre                       0
lyrics                      0
len                         0
dating                      0
violence                    0
world/life                  0
night/time                  0
shake the audience          0
family/gospel               0
romantic                    0
communication               0
obscene                     0
music                       0
movement/places             0
light/visual perceptions    0
family/spiritual            0
sadness                     0
feelings                    0
topic                       0
age                         0
dtype: int64


**No null values exist across this entire dataset.**

In [20]:
#Get column data types (dtype), overall count of each present dtype, and memory usage
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28362 entries, 0 to 28361
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                28362 non-null  int64  
 1   artist_name               28362 non-null  object 
 2   track_name                28362 non-null  object 
 3   release_date              28362 non-null  int64  
 4   genre                     28362 non-null  object 
 5   lyrics                    28362 non-null  object 
 6   len                       28362 non-null  int64  
 7   dating                    28362 non-null  float64
 8   violence                  28362 non-null  float64
 9   world/life                28362 non-null  float64
 10  night/time                28362 non-null  float64
 11  shake the audience        28362 non-null  float64
 12  family/gospel             28362 non-null  float64
 13  romantic                  28362 non-null  float64
 14  commun